AWSCognitoAuth/Internal/UICKeyChainStore/AWSCognitoAuthUICKeyChainStore.m

```diff
@@ -933,6 +933,26 @@ + (NSArray *)prettify:(CFTypeRef)itemClass items:(NSArray *)items
 
 #pragma mark -
 
+- (void)migrateToCurrentAccessibility {
+    NSArray *items = [self allItems];
+    for (NSDictionary *item in items) {
+        CFComparisonResult result = CFStringCompare((CFStringRef)item[@"accessibility"],
+                                                    [self accessibilityObject], 0);
+        if (result == 0) {
+            continue;
+        }
+        NSString *key = item[@"key"];
+        NSObject *value = item[@"value"];
+        if ([value isKindOfClass:[NSString class]]) {
+            [self setString: (NSString *)value forKey:key];
+        } else {
+            [self setData: (NSData *)value forKey:key];
```

Comment:
Are we trusting external input here? Or do we control 100% the type of `value`? If not, should we add an additional `kindOfClass` check for NSData to avoid surprises?

Reply:
Ideally these values should not be changed outside the library. But the developer can recreate the same keychain using the unique identifier and right to the same location. So to be on the safer side, I added check for NSData as well.